lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -158,6 +158,26 @@ function dosomething_northstar_get_northstar_user($drupal_id) {
 }
 
 /**
+ * Get a Northstar user for a given index & type. Useful for non Drupal ID lookups.
+ * @param  string $index_type The type of index to query for (Email, Facebook, Mobile, _id)
+ * @param  string $index The index to query for
+ * @return object|null
+ */
+function dosomething_northstar_get_user_by_custom_index($index_type, $index) {
```

Comment:
@angaither great, merged the two function & tested to make sure its still caching alright. Going to run through a bit more testing but let me know if this looks good!